### PR TITLE
a8n: Persist GitHub ChangesetEvents to database

### PR DESCRIPTION
**Pull request by tsenart**
This adds the necessary methods to the `a8n.Store` to upsert, list and count `a8n.ChangesetEvents`.

After loading changesets in the `a8n.Syncer` we upsert them into the database, making sure that on conflict they're updated.

The `key` column on `changeset_events` is added in preparation for the introduction of webhooks, where we could potentially receive the same event via API calls and via webhook payloads. In that case, we want to make sure that we can reconcile the two different representations of the same event and update the database entry to its newest version.

The PR also adds a minimal first GraphQL API for changeset events. It's minimal, because we still need to work out the schema.

cc @felixfbecker

### CODE REVIEW

**Comment by tsenart** [comment]
This is a comment.

**Comment by tsenart** [comment]
Making this PR the playground for fetching timeline events from Github.

File: enterprise/pkg/a8n/store.go [hunk 1]
```diff
@@ -427,6 +427,236 @@ func (s *Store) updateChangesetsQuery(cs []*a8n.Changeset) (*sqlf.Query, error)
...
+	)
+}
+
+// CountChangesetEventsOpts captures the query options needed for
+// counting changesets.
+type CountChangesetEventsOpts struct {
+	ChangesetID int64
+}
+
+// CountChangesetEvents returns the number of changesets in the database.
```
Comment (mrnugget) [inline enterprise/pkg/a8n/store.go]:
```suggestion
// CountChangesetEvents returns the number of changeset events in the database.
```

File: enterprise/pkg/a8n/store.go [hunk 2]
```diff
@@ -427,6 +427,236 @@ func (s *Store) updateChangesetsQuery(cs []*a8n.Changeset) (*sqlf.Query, error)
...
+
+	return sqlf.Sprintf(
+		listChangesetEventsQueryFmtstr,
+		sqlf.Join(preds, "\n AND "),
+		opts.Limit,
+	)
+}
+
+// CountChangesetEventsOpts captures the query options needed for
+// counting changesets.
```
Comment (mrnugget) [inline enterprise/pkg/a8n/store.go]:
```suggestion
// counting changeset events.
```

**Pull request merged**